src/QiParsers.h

```diff
@@ -52,6 +52,7 @@ inline bool parseNumber(char decimalMark, char groupingMark, Iterator& first,
   double sum = 0, denom = 1;
   NumberState state = STATE_INIT;
   bool seenNumber = false;
+  bool negative = *first == '-';
```

Comment:
Actually, I think this would be nicer as `int sign = 1`, and then change it in the `*cur == '-'` block below. Then you return `sign * sum`